Description: "Fix" handling of 'meta8' option for non-latin1 locales
 This patch was unapplied because even the author admits that it is a hack.
 Further, it hardcodes UTF-8.
Author: amateur <tianlei.zhao@gmail.com>
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=558725#5
Bug-Debian: http://bugs.debian.org/558725
Forwarded: no
Reviewed-By: Unapplied by Ryan Kavanagh <ryanakca@kubuntu.org>
Last-Update: 2011-03-06

--- rxvt-unicode-9.09.orig/src/command.C
+++ rxvt-unicode-9.09/src/command.C
@@ -810,8 +810,21 @@ rxvt_term::key_press (XKeyEvent &ev)
             {
               char *ch;
 
-              for (ch = kbuf; ch < kbuf + len; ch++)
-                *ch |= 0x80;
+              for (ch = kbuf; ch < kbuf + len; ch++) {
+                /* UTF-8 encoding */
+                char byte2 = 0x80 | (*ch  & 0x3f);
+                *ch = 0xC0 | (0x80 >> 6) | ((*ch >> 6) & 0x03);
+                ch ++;
+
+                char * current = NULL;
+                char * last = kbuf + len;
+                for (current=last; current > ch; current--) {
+                    *current = *(current-1);
+                }
+
+                len ++;
+                *ch = byte2;
+              }
 
               meta = 0;
             }
